libr/asm/arch/8051/8051_ass.c

```diff
@@ -458,13 +458,15 @@ static bool address_direct(char const* addr_str, ut8* addr_out) {
 		found = true;
 	}
 
-	if (!found) {
+	/* need opinion in order to remove this comment
```

Comment:
use #if 0 to comment code

Reply:
sgtm, to check for reg on immediate, but can you provide a example string for me to understand the case you are aiming to handle here?

Reply:
good to remove or to keep?

Reply:
All my draft PR was about this question. I think I had a bad idea because ther ecould not be both register and operand parsing in this architecture. See https://www.keil.com/support/man/docs/is51/is51_opcodes.htm

Reply:
Keep it with if0, and address the other reviews